File: src/main/java/io/transwarp/logparser/filter/Filter.java
package io.transwarp.logparser.filter;

import io.transwarp.logparser.util.Record;

/**
 * Author: stk
 * Date: 18/3/27
 */
public interface Filter {
    /**
     * Filter the record.
     *
     * @param record one record
     * @return if the record needs to be save into file, then return true.
     */
    boolean filter(Record record);

    /**
     * Clean status.
     */
    void clean();
}

File: src/main/scala/io/transwarp/logparser/filter/Filter.scala
package io.transwarp.logparser.filter

import io.transwarp.logparser.util.LogEntry

/**
  * Author: stk
  * Date: 2018/4/3
  */
abstract class Filter {
  val params: List[String]

  /**
    * Filter log entry.
    *
    * @param logEntry log entry
    * @return if the log entry needs to be saved, then return true, otherwise false.
    */
  def filter(logEntry: LogEntry): Boolean

  def copy(): Filter
}
